foundationdb/src/transaction.rs

```diff
@@ -811,6 +825,36 @@ impl Transaction {
         )
     }
 
+    /// Commits the transaction if any write has been performed. Otherwise, returns a future that will
+    /// be immediately ready with a `TransactionCommitted` value without calling the underlying FoundationDB
+    /// API.
+    /// 
+    /// This behaviour addresses performance issues for high-concurrency applications, where no-op transactions
+    /// are frequently created and committed, which can result in large tail latencies.
+    /// 
+    /// More info can be found in the relevant [pull request](https://github.com/foundationdb-rs/foundationdb-rs/pull/334)
+    pub fn commit_if_needed(self) -> Pin<Box<dyn Future<Output = TransactionResult> + Send + Sync + Unpin>> {
+        if !self.should_commit() {
```

Comment:
This is semantically wrong: it returns a `TransactionCommitted` even when no commit happened.
That leaks APIs like [TransactionCommitted::committed_version()](https://docs.rs/foundationdb/latest/foundationdb/struct.TransactionCommitted.html#method.committed_version), which are invalid without a real commit, and misleads `db.run` users.